Dockerfile.windows

```diff
@@ -250,18 +256,89 @@ RUN `
   Remove-Item C:\gitsetup.zip; `
   `
   Write-Host INFO: Creating source directory...; `
-  New-Item -ItemType Directory -Path ${GOPATH}\src\github.com\docker\docker | Out-Null; `
+  #TODO - Per https://docs.docker.com/engine/reference/builder/ 
+  #       the below line is redundant and should be removed.
+  #New-Item -ItemType Directory -Path ${GOPATH}\src\github.com\docker\docker | Out-Null; `
```

Comment:
We should update those docs a bit. Although `WORKDIR` _will_ create the directory if it doesn't exist, when it _does_ it always creates the directory as `root` (or likely the equivalent on Windows).

This means that if the Dockerfile is using a different `USER` (either during build, or when running the container, so with `--user=<some-user>:<some-group>`, hence in the following:

```Dockerfile
FROM foo
USER some_non_privileged_user
WORKDIR /some/path/
```

`/some/path/` will be owned by `root`, and not `some_non_privileged_user`

For that reason, it's often good practice to create the directory manually, instead of relying on the side effect of `WORKDIR` creating the directory.

Reply:
Changed to create the directory explicitly and removed the WORKDIR directive